src/internals/objects/Objects.ts

```diff
@@ -260,4 +261,39 @@ export namespace Objects {
     union extends string | _ | unset = unset,
     value = unset
   > = Functions.PartialApply<RecordFn, [union, value]>;
+
+  type JoinPath<A extends string, B extends string> = [A] extends [never]
+    ? B
+    : [B] extends [never]
+    ? A
+    : `${A}.${B}`;
+
+  type AllPathsImpl<T> = T extends Primitive
+    ? never
+    : unknown extends T
+    ? string
+    : T extends any[]
+    ? Call<Tuples.Keys, T> extends infer key extends string | number
+      ? `${key}` | JoinPath<`${key}`, AllPathsImpl<T[number]>>
+      : never
+    : keyof T extends infer key extends keyof T & string
+    ? key extends any
+      ? key | JoinPath<key, AllPathsImpl<T[key]>>
+      : never
+    : never;
```

Comment:
not required for this PR, but It might make sense to start puting object function implementations under a impl/objects.ts file